core/src/main/java/bisq/core/trade/protocol/tasks/ProcessInitMultisigRequest.java

```diff
@@ -122,8 +122,8 @@ protected void run() {
             log.info("Importing exchanged multisig hex for trade {}", trade.getId());
             MoneroMultisigInitResult result = multisigWallet.exchangeMultisigKeys(Arrays.asList(peers[0].getExchangedMultisigHex(), peers[1].getExchangedMultisigHex()), xmrWalletService.getWalletPassword());
             processModel.setMultisigAddress(result.getAddress());
-            processModel.getProvider().getXmrWalletService().saveMultisigWallet(trade.getId()); // save multisig wallet once it's created
             trade.setStateIfValidTransitionTo(Trade.State.MULTISIG_COMPLETED);
+            trade.saveWallet(); // save multisig wallet on completion
```

Comment:
```suggestion
            xmrWalletService.saveMultisigWallet(trade.getId()); // save multisig wallet on completion
            trade.setStateIfValidTransitionTo(Trade.State.MULTISIG_COMPLETED);
```

Reply:
Why this suggestion? Essentially a trade encapsulates its multisig wallet, so trade.saveWallet() is more encapsulated, and calls out to the XmrWalletService.saveMultisigWallet(tradeId) for implementation. Perhaps the comment should read, "save trade wallet on completion"